plugins/module_utils/crypto/cryptography_openssh.py

```diff
@@ -0,0 +1,484 @@
+
+                if passphrase_validation_error:
+                    raise InvalidPassphraseError("%s cannot be encoded from text" % passphrase)
+
+                self.__encryption_algorithm = serialization.BestAvailableEncryption(passphrase.encode(encoding=_TEXT_ENCODING))
+            else:
+                self.__encryption_algorithm = serialization.NoEncryption()
+
+            if path:
+                self.__keytype = None
+                self.__load_keypair(path, passphrase)
+            else:
+                if keytype not in self.__algorithm_parameters.keys():
+                    raise InvalidKeyTypeError("%s is not a valid keytype. Valid keytypes are %s" % (keytype, ", ".join(self.__algorithm_parameters.keys())))
+
+                self.__keytype = keytype
+
+                if size:
+                    self.__size = size
+                else:
+                    self.__size = self.__algorithm_parameters[self.__keytype]['default_size']
+
+                self.__generate_keypair()
+
+        @property
+        def PrivateKey(self):
+            """Returns the openssh formatted private key of this key pair
+            """
+            return self.__privatekey
+
+        @property
+        def PublicKey(self):
+            """Returns the openssh formatted public key of this key pair
+            """
+            return self.__publickey
+
+        @property
+        def Size(self):
+            """Returns the size of the private key of this key pair
+            """
+            return self.__size
+
+        @property
+        def KeyType(self):
+            """Returns the key type of this key pair
+            """
+            return self.__keytype
+
+        @property
+        def EncryptionAlgorithm(self):
+            """Returns the key encryption algorithm of this key pair
+            """
+            return self.__encryption_algorithm
+
+        def Sign(self, data):
+            """Returns bytes of data signed with the private key of this key pair
+
+               :data: byteslike data to sign
+            """
+            signer_params = {
+                'rsa': {
+                    'padding': padding.PSS(
+                        mgf=padding.MGF1(hashes.SHA256()),
+                        salt_length=padding.PSS.MAX_LENGTH
+                    ),
+                    'algorithm': hashes.SHA256(),
+                },
+                'dsa': {
+                    'algorithm': hashes.SHA256(),
+                },
+                'ecdsa': {
+                    'signature_algorithm': ec.ECDSA(hashes.SHA256()),
+                },
+                'ed25519': {},
+            }
+
+            try:
+                signed_data = self.__privatekey.sign(
+                    data,
+                    **signer_params[self.__keytype]
+                )
+            except TypeError as e:
+                raise InvalidDataError(e)
+
+            return signed_data
+
+        def Update_Passphrase(self, passphrase):
```

Comment:
Method names should follow the same convention as property names:
```suggestion
        def update_passphrase(self, passphrase):
```